### DISCUSSION TITLE
Companion APP in the xDrip does not work with Medtronic Guardian with sensor 3 or 4.

### DISCUSSION THREAD

**Discussion by pronunta** [NightscoutFoundation/xDrip#2999] [Ideas]
Companion APP in the xDrip does not work.
Use Medtronic Guardian with sensor 3 or 4.
My son has AAPS, but Carelink doesn't work :(

**Comment by pronunta** [comment]
![7](https://github.com/NightscoutFoundation/xDrip/assets/67656669/1c3a8cd2-059d-4043-b68f-e57c2db471b8)

**Comment by pronunta** [comment]
![5](https://github.com/NightscoutFoundation/xDrip/assets/67656669/e325aca7-67fc-4af3-b7e1-b745d06ec24a)

**Comment by Navid200** [comment]
What version of xDrip are you using?

**Reply by pronunta** [reply]
Latest version 
Nightly build 10th Aug 202
![3](https://github.com/NightscoutFoundation/xDrip/assets/67656669/5ff26094-a201-42cb-ae78-17a3434ea083)

**Reply by Navid200** [reply]
Please post a screenshot of the main screen.

**Reply by pronunta** [reply]
the older BG values are from nightscout follower. If i comute to Companion App the BG is unavalaible
![9](https://github.com/NightscoutFoundation/xDrip/assets/67656669/7cb96577-bffe-482f-a60e-c6859f37e632)

**Comment by Navid200** [comment]
Have you given xDrip notification access?
https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by pronunta** [reply]
Of course! xDrip has notifications acces and the aaps notifications are off to not create any problems.

**Reply by Navid200** [reply]
Do you mind posting a screenshot showing notification access?

**Reply by pronunta** [reply]
![10](https://github.com/NightscoutFoundation/xDrip/assets/67656669/1c8690db-562c-49bf-a466-541f13b2bb47)
![11](https://github.com/NightscoutFoundation/xDrip/assets/67656669/746b69a1-251e-4e60-950d-47d0b7290dbb)

**Comment by nlof** [comment]
I don't use Medtronic but FYI there were several posts in a diabetes group that Carelink on Android is down for several days now. Specifically in Romania, and that's where it seems that OP is from. Also, it supposedly works on iPhones.

UPDATE: Oh I see, because Carelink is down, you tried companion app on your son's phone in order to get the readings into Xdrip. And from xdrip you could send it to AAPS and to other phones, even nightscout, right?
QUESTION: How were you sending values to nightscout if Carelink is down? Maybe you could PM me directly as this is off topic.

**Reply by Mishu12111** [reply]
This is how you can send values to nightscout from xDrip+ ,if that is what you are asking. 
![WhatsApp Image 2023-08-21 at 13 38 05](https://github.com/NightscoutFoundation/xDrip/assets/89389790/ce62d1bd-da33-4393-a874-bf7929a3fc3b)
![WhatsApp Image 2023-08-21 at 13 38 06 (1)](https://github.com/NightscoutFoundation/xDrip/assets/89389790/fa23c4e8-c7ba-4a5e-b83e-7e8885be59ca)
![WhatsApp Image 2023-08-21 at 13 38 06](https://github.com/NightscoutFoundation/xDrip/assets/89389790/45204581-6df6-43a8-93e1-d24eb688368f)

**Reply by nlof** [reply]
Mishu, I see there is no PM option so I'll reply here.
OP seemed to have been able to upload from Guardian 3 directly to Nightscout somehow without a functioning Carelink/ Xdrip. That's what I was asking about maybe in a convoluted way...
I know how to upload to NS once BG's reach xdrip from another source (such as Carelink), that's piece of cake 🙂

**Reply by Navid200** [reply]
The companion app mode let's you get readings from another app running on the same phone as long as the app creates notifications for its readings.
This explains Companion app mode:
https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by nlof** [reply]
Thanks Navid, I know that too. Apparently OP can't get that to work. I had also issues with companion app when I tried it a while ago with Dex1. I haven't tried it since because I'm using Anubis which works with G6 BYODA.

**Reply by Navid200** [reply]
Companion app mode is a relatively new feature (may be even less than a year ) in xDrip and the lead developer has been improving it ever since.  
Looking at the release notes, the last improvement was made only 2 days ago.
https://navid200.github.io/xDrip/docs/ReleaseNotes.html

It takes a lot of time and effort to keep xDrip going.

It's a little bit unfair to try something and then post somewhere and say it doesn't work.  For all you know, it may work now.
There is no way we can test every single available handset out there.
Why do you think Dexcom only has a few devices on their approved list?
Yet, you can use almost any device to collect directly from G6 and many other devices.
You have chosen not to use xDrip for that.  And that is fine too.  You can do whatever you like.

The Dexcom One app does not create notifications including glucose readings.  That is a requirement for the Companion app mode.  So, I am not surprised that it did not work for you.  
But, why are we talking about that in this discussion?

I hope these discussions can be fruitful letting users find answers and letting us know what needs attention.

**Reply by nlof** [reply]
I am sorry to have sidetracked the conversation. I wanted to do it all via PM which apparently doesn't exist.  I just wanted to know from the OP how was he sending info from Guardian directly to NS without Carelink or Xdrip.
While trying to clarify this I replied to both you and Mishu but that entire exchange is totally irrelevant to the guardian companion app issue. While I did try the companion app long ago and back then it was suggested that I do what I did, in the end things got resolved another way. 
I wasn't criticizing Xdrip in any shape or form, it works fine in my current setup with BYODA G6.
I just wish OP could answer my original question and I don't see a way to PM him.
Regarding the companion app for Guardian I saw the thread that tries to fix it...

**Reply by Navid200** [reply]
That's actually a very good question.  I am sorry I misunderstood and thanks for clarifying.

**Comment by ah7675** [comment]
https://github.com/NightscoutFoundation/xDrip/discussions/3025#discussion-5524281
FYI
